src/pages/YearPickerPage.js

```diff
@@ -70,9 +70,10 @@ class YearPickerPage extends React.Component {
      * @param {String} text
      */
     filterYearList(text) {
+        const searchText = text.replace(/\D/g, '');
         this.setState({
```

Comment:
NAB: You don't have to perform search if the search text has not been changed. We can check if the new search text is same as the current value.

```
        this.setState((prevState) => {
         if (searchText === prevState.inputText) {
          return {};
         } else {
           inputText: searchText,
            yearOptions: _.filter(this.yearList, year => year.text.includes(searchText)),
         }
       })
```